### PR TITLE
SSL updates

### PR DESCRIPTION
**Pull request by nicktrav**
Updates SSL behavior for `vertica-python` to:
- fix issue with sockets blocking when an SSL connection is used (uber/vertica-python#96)
- allow for a customizable SSL connection by passing in an arbitrary `ssl.SSLContext` to the connection_info dict

### CODE REVIEW

**Comment by rileycrane** [comment]
Please add this

**Comment by nicktrav** [comment]
Bump. @alexjikim - does this look ok to merge in?

**Comment by alexjikim** [comment]
im a bit concerned about this due to the fact that this will break clients on older versions of python 2.7.x using an outdated ssl module.

Im considering pulling the ssl import into the 'if using ssl' code block so only clients that actually use ssl will need an up to date python. Let me know if you have concerns about this approach.

**Comment by nicktrav** [comment]
Sounds reasonable. Do you mean something like:
```python
if ssl_options is not None and ssl_options is not False:
  from ssl import CertificateError, SSLError
  ...
```
Happy to amend the commit.

**Comment by alexjikim** [comment]
yea. if you could add that it would be great. ill merge afterwards.

File: vertica_python/vertica/connection.py [hunk 1]
```diff
@@ -99,24 +99,32 @@ def _socket(self):
...
-            raw_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            raw_socket.connect((self.options['host'], self.options['port']))
+        host = self.options.get('host')
+        port = self.options.get('port')
+        raw_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        raw_socket.connect((host, port))
+
+        ssl_options = self.options.get('ssl')
+        if ssl_options is not None and ssl_options is not False:
+            from ssl import CertificateError, SSLError
```
Comment (nicktrav) [inline vertica_python/vertica/connection.py]: @alexjikim - made the change, as suggested

Comment (nicktrav) [inline vertica_python/vertica/connection.py]: @alexjikim - bump.

**Comment by Bennyelg** [comment]
Better and more acquire to write is:
```python
if ssl_options is not None and  not ssl_options: ...
```

**Comment by nicktrav** [comment]
@Bennyelg - made this change.

@alexjikim - bumping again. We're have this forked internally, so would be good to get this into master. Cheers.

**Comment by alexjikim** [comment]
sorry for the delay, ive been out of office and dont have the means to test or tag a new release. I can get to this next week when im back.

**Comment by alexjikim** [comment]
in pypi as 0.5.7

**Pull request merged**